igvc_navigation/src/path_follower/Path_follower.cpp

```diff
@@ -1,37 +1,64 @@
 #define _USE_MATH_DEFINES
 
+#include <Eigen/Dense>
+#include <cmath>
+#include <iostream>
+
 #include <geometry_msgs/PointStamped.h>
 #include <geometry_msgs/PoseStamped.h>
 #include <igvc_msgs/velocity_pair.h>
 #include <nav_msgs/Odometry.h>
-#include <nav_msgs/Path.h>
-#include <ros/ros.h>
+
 #include <tf/transform_datatypes.h>
-#include <Eigen/Dense>
-#include <cmath>
+
 #include <igvc_utils/NodeUtils.hpp>
 #include <igvc_utils/RobotState.hpp>
-#include <iostream>
-#include "SmoothControl.h"
 
-ros::Publisher cmd_pub;
-ros::Publisher target_pub;
-ros::Publisher trajectory_pub;
+#include "Path_follower.h"
 
-nav_msgs::PathConstPtr path;
-geometry_msgs::PointStampedConstPtr waypoint;
+Path_follower::Path_follower()
+{
+  ros::NodeHandle nh;
+  ros::NodeHandle pNh("~");
+
+  // load controller parameters
+  double target_velocity;
+  double axle_length;
+  double k1;
+  double k2;
+  double granularity;
+  double lookahead_dist;
+  igvc::param(pNh, "target_v", target_velocity, 1.0);
+  igvc::param(pNh, "axle_length", axle_length, 0.52);
+  igvc::param(pNh, "k1", k1, 1.0);
+  igvc::param(pNh, "k2", k2, 3.0);
+  igvc::param(pNh, "granularity", granularity, 2.0);
```

Comment:
`granularity` should be much higher by default,  at least in the 100's range.  Keep in mind this parameter dictates the quality of the trajectory visualization; with a `granularity` of 2, the trajectory would be visualized as two connected line segments.